beetsplug/subsonic.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+# Subsonic plugin for Beets
+# Allows to trigger a library scan on Subsonic
+# after music has been imported with Beets
+# Your Beets configuration file should contain
+# a subsonic section like the following:
+#   subsonic:
+#       host: 192.168.x.y (the IP address where Subsonic listens on)
+#       port: 4040 (default)
+#       user: <your username>
+#       pass: <your password>
+# Plugin wrote by maffo999
+
+"""Updates Subsonic library on Beets import
+"""
+from __future__ import division, absolute_import, print_function
+
+from beets.plugins import BeetsPlugin
+import requests
+import string
+import hashlib
+import random
+
+
+class Subsonic(BeetsPlugin):
+    def __init__(self):
+        super(Subsonic, self).__init__()
+        self.register_listener('import', self.loaded)
+
+    def loaded(self):
+        host = self.config['host'].get()
+        port = self.config['port'].get()
+        user = self.config['user'].get()
+        passw = self.config['pass'].get()
+        r = string.ascii_letters + string.digits
+        url = "http://"+str(host)+":"+str(port)+"/rest/startScan"
+        salt = "".join([random.choice(r) for n in range(6)])
+        t = passw + salt
+        token = hashlib.md5()
+        token.update(t.encode('utf-8'))
```

Comment:
Consider breaking up these lines of code and adding some comments to say what you're doing: generating a salt, with which you hash a password to send to the server.